### ISSUE TITLE
Property was accessed through the prototype chain. Use 'decorate' instead

### ISSUE THREAD

**Issue by LeonardoGentile** [mobxjs/mobx#1506]
Hello,
since MobX 4 I'm (sometimes) having the following issue.
I'm declaring a computed property and an autorun attached to a React component, like this:

```
@inject(routerStore)
class RouteNode extends Component {
  
  // .....  

  @computed get isIntersection() {
    return this.routerStore.intersectionNode === this.nodeName;
  }

  componentDidMount() {
    this.autorunDisposer = autorun(() => {
      if (this.isIntersection) {
        this.setState({
          route: this.routerStore.route
        });
      }
    });
  }

  componentWillUnmount() {
    this.autorunDisposer();
  }
  
  // ....
}
```

Sometimes I receive this in the console:

```
Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Autorun@163] Error: [mobx] Property 'isIntersection' of '[object Object]' was accessed through the prototype chain. Use 'decorate' instead to declare the prop or access it statically through it's owner
```

So probably I should not declare `@computed get isIntersection()` on the component class but somehow use decorate?

The issue seems similar to #1427 but it's not exactly clear to me how I could solve it in this case 🤔

**Comment by urugator** [comment]
I can't replicate https://codesandbox.io/s/rl5z5xno9p 
Can you set up a reproduction? 
Do you extend from a custom Component or something like that?

**Comment by mweststrate** [comment]
HMR?

Op do 19 apr. 2018 19:06 schreef urugator <notifications@github.com>:

**Comment by LeonardoGentile** [comment]
Full implementation [here](https://github.com/LeonardoGentile/react-mobx-router5/blob/master/src/modules/routeNode.js)

@mweststrate yes indeed this happen during HMR!

**Comment by AmazingTurtle** [comment]
I'm having the same errors with HMR.
Using react-hot-loader@4.1.2, mobx@4.2.0, mobx-react@5.0.0.

Following babel plugins:
`"plugins": [
    [
      "import",
      {
        "libraryName": "antd",
        "libraryDirectory": "lib",
        "style": "css"
      }
    ],
    "transform-runtime",
    "lodash",
    "transform-decorators-legacy",
    "extract-hoc/babel",
    "react-hot-loader/babel"
  ]`

**Comment by mweststrate** [comment]
decorators and react hot loader has always been troublesome, and I gave up on it a few years ago :). That being said, if changing the above error to a warning fixes the problems, a PR for that will probably be accepted

**Comment by mweststrate** [comment]
Released as 4.3.0
